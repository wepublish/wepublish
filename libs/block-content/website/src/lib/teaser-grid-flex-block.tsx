import {css, styled} from '@mui/material'
import {BuilderTeaserGridFlexBlockProps, useWebsiteBuilder} from '@wepublish/website/builder'
import {
  Block,
  FlexAlignment,
  Teaser as TeaserType,
  TeaserGridFlexBlock as TeaserGridFlexBlockType
} from '@wepublish/website/api'
import {isImageBlock} from './image-block'
import {isTitleBlock} from './title-block'

export const isTeaserGridFlexBlock = (block: Block): block is TeaserGridFlexBlockType =>
  block.__typename === 'TeaserGridFlexBlock'

export const TeaserGridFlexBlockWrapper = styled('div')`
  display: grid;
  gap: ${({theme}) => theme.spacing(2)};
  grid-template-columns: 1fr;
  align-items: stretch;

  ${({theme}) => css`
    ${theme.breakpoints.up('sm')} {
      grid-template-columns: 1fr 1fr;
    }

    ${theme.breakpoints.up('md')} {
      grid-template-columns: repeat(12, 1fr);
    }
  `}
`

const TeaserWrapper = styled('article')<FlexAlignment>`
  ${({theme, h, w, x, y}) => css`
    ${theme.breakpoints.up('md')} {
      grid-column-start: ${x + 1};
      grid-column-end: ${x + 1 + w};
      grid-row-start: ${y + 1};
      grid-row-end: ${y + 1 + h};
    }
  `}
`

const TeaserInnerWrapper = styled('div')`
  display: grid;
  gap: ${({theme}) => theme.spacing(1)};
  grid-template-rows: auto;
  grid-auto-rows: max-content;
`

const teaserTitle = (teaser: TeaserType) => {
  switch (teaser.__typename) {
    case 'PageTeaser': {
      const titleBlock = teaser.page?.blocks.find(isTitleBlock)
      return teaser.title || titleBlock?.title || teaser.page?.title
    }

    case 'ArticleTeaser': {
      const titleBlock = teaser.article?.blocks.find(isTitleBlock)
      return teaser.title || titleBlock?.title || teaser.article?.title
    }

    case 'CustomTeaser':
      return teaser.title
  }
}

const teaserPreTitle = (teaser: TeaserType) => {
  switch (teaser.__typename) {
    case 'PageTeaser':
      return teaser.preTitle
    case 'ArticleTeaser':
      return teaser.preTitle || teaser.article?.preTitle
    case 'CustomTeaser':
      return teaser.preTitle
  }
}

const teaserLead = (teaser: TeaserType) => {
  switch (teaser.__typename) {
    case 'PageTeaser': {
      const titleBlock = teaser.page?.blocks.find(isTitleBlock)
      return teaser.lead || titleBlock?.lead || teaser.page?.description
    }

    case 'ArticleTeaser': {
      const titleBlock = teaser.article?.blocks.find(isTitleBlock)
      return teaser.lead || titleBlock?.lead || teaser.article?.lead
    }

    case 'CustomTeaser':
      return teaser.lead
  }
}

const teaserUrl = (teaser: TeaserType) => {
  switch (teaser.__typename) {
    case 'PageTeaser': {
      return teaser.page?.url
    }

    case 'ArticleTeaser': {
      return teaser.article?.url
    }

    case 'CustomTeaser':
      return teaser.contentUrl
  }
}

const teaserImage = (teaser: TeaserType) => {
  switch (teaser.__typename) {
    case 'PageTeaser': {
      const imageBlock = teaser.page?.blocks.find(isImageBlock)
      return teaser.image ?? imageBlock?.image
    }

    case 'ArticleTeaser': {
      const imageBlock = teaser.article?.blocks.find(isImageBlock)
      return teaser.image ?? imageBlock?.image
    }

    case 'CustomTeaser':
      return teaser.image
  }
}

const TeaserImage = styled('img')`
  object-fit: cover;
  height: 100%;
`

const TeaserTitleWrapper = styled('header')``

export type TeaserProps = {
  teaser?: TeaserType | null
  alignment: FlexAlignment
  className?: string
  showLead?: boolean
}

export const Teaser = ({teaser, alignment, className, showLead}: TeaserProps) => {
  const title = teaser && teaserTitle(teaser)
  const preTitle = teaser && teaserPreTitle(teaser)
  const lead = teaser && teaserLead(teaser)
  const href = (teaser && teaserUrl(teaser)) ?? ''
  const image = teaser && teaserImage(teaser)

  const {
    elements: {H5, H6, Link}
  } = useWebsiteBuilder()

  return (
    <TeaserWrapper {...alignment}>
      <Link color="inherit" underline="none" href={href}>
        <TeaserInnerWrapper className={className}>
          {image?.url && (
            <TeaserImage
              src={image.url}
              width={image.width}
              height={image.height}
              alt={image.description ?? ''}
            />
          )}

          <TeaserTitleWrapper>
            <H5 component="h1">
              {preTitle && `${preTitle}: `} {title}
            </H5>
          </TeaserTitleWrapper>

          {showLead && <H6 component="p">{lead}</H6>}
        </TeaserInnerWrapper>
      </Link>
    </TeaserWrapper>
  )
}

export const TeaserGridFlexBlock = ({flexTeasers, className}: BuilderTeaserGridFlexBlockProps) => {
  return (
    <TeaserGridFlexBlockWrapper className={className}>
      {flexTeasers?.map((teaser, index) => (
        <Teaser key={index} {...teaser} />
      ))}
    </TeaserGridFlexBlockWrapper>
  )
}
