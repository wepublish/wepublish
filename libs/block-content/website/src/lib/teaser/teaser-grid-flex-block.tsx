import {css, styled} from '@mui/material'
import {Block, TeaserGridFlexBlock as TeaserGridFlexBlockType} from '@wepublish/website/api'
import {BuilderTeaserGridFlexBlockProps, useWebsiteBuilder} from '@wepublish/website/builder'

export const isTeaserGridFlexBlock = (block: Block): block is TeaserGridFlexBlockType =>
  block.__typename === 'TeaserGridFlexBlock'

export const TeaserGridFlexBlockWrapper = styled('div')`
  display: grid;
  gap: ${({theme}) => theme.spacing(2)};
  grid-template-columns: 1fr;
  align-items: stretch;

  ${({theme}) => css`
    ${theme.breakpoints.up('sm')} {
      grid-template-columns: 1fr 1fr;
    }

    ${theme.breakpoints.up('md')} {
      grid-template-columns: repeat(12, 1fr);
    }
  `}
`

export const TeaserGridFlexBlock = ({
  flexTeasers,
  showLead,
  className
}: BuilderTeaserGridFlexBlockProps) => {
  const {
    blocks: {Teaser}
  } = useWebsiteBuilder()

  return (
    <TeaserGridFlexBlockWrapper className={className}>
      {flexTeasers?.map((teaser, index) => (
        <Teaser key={index} showLead={showLead} {...teaser} />
      ))}
    </TeaserGridFlexBlockWrapper>
  )
}
