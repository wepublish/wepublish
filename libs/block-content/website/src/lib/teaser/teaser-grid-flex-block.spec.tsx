import {render} from '@testing-library/react'
import * as stories from './teaser-grid-flex-block.stories'
import {composeStories} from '@storybook/react'

const storiesCmp = composeStories(stories)

describe('Teaser Grid Flex Block', () => {
  Object.entries(storiesCmp).forEach(([story, Component]) => {
    it(`should render ${story}`, () => {
      const {asFragment} = render(<Component />)
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
