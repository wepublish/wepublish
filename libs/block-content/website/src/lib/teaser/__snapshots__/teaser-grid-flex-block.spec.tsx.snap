// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teaser Grid Flex Block should render Default 1`] = `
<DocumentFragment>
  <div
    class="css-dkdiq6"
  >
    <article
      class="css-1p0ks0j"
      x="0"
      y="0"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
               Teambesprechung vom 23.05.
            </h1>
          </header>
        </div>
      </a>
    </article>
    <article
      class="css-1aj06cw"
      x="3"
      y="0"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title:  Title
            </h1>
          </header>
        </div>
      </a>
    </article>
    <article
      class="css-hgavva"
      x="7"
      y="0"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title:  Title
            </h1>
          </header>
        </div>
      </a>
    </article>
    <article
      class="css-2t6kni"
      x="7"
      y="8"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
               Teambesprechung vom 23.05.
            </h1>
          </header>
        </div>
      </a>
    </article>
    <article
      class="css-swar48"
      x="0"
      y="4"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title:  Title
            </h1>
          </header>
        </div>
      </a>
    </article>
    <article
      class="css-1oxqzs1"
      x="3"
      y="6"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title:  Title
            </h1>
          </header>
        </div>
      </a>
    </article>
  </div>
</DocumentFragment>
`;

exports[`Teaser Grid Flex Block should render WithClassName 1`] = `
<DocumentFragment>
  <div
    class="extra-classname css-dkdiq6"
  >
    <article
      class="css-1p0ks0j"
      x="0"
      y="0"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
               Teambesprechung vom 23.05.
            </h1>
          </header>
        </div>
      </a>
    </article>
    <article
      class="css-1aj06cw"
      x="3"
      y="0"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title:  Title
            </h1>
          </header>
        </div>
      </a>
    </article>
    <article
      class="css-hgavva"
      x="7"
      y="0"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title:  Title
            </h1>
          </header>
        </div>
      </a>
    </article>
    <article
      class="css-2t6kni"
      x="7"
      y="8"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
               Teambesprechung vom 23.05.
            </h1>
          </header>
        </div>
      </a>
    </article>
    <article
      class="css-swar48"
      x="0"
      y="4"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title:  Title
            </h1>
          </header>
        </div>
      </a>
    </article>
    <article
      class="css-1oxqzs1"
      x="3"
      y="6"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title:  Title
            </h1>
          </header>
        </div>
      </a>
    </article>
  </div>
</DocumentFragment>
`;

exports[`Teaser Grid Flex Block should render WithEmotion 1`] = `
<DocumentFragment>
  <div
    class="css-dkdiq6"
  >
    <article
      class="css-1p0ks0j"
      x="0"
      y="0"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
               Teambesprechung vom 23.05.
            </h1>
          </header>
        </div>
      </a>
    </article>
    <article
      class="css-1aj06cw"
      x="3"
      y="0"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title:  Title
            </h1>
          </header>
        </div>
      </a>
    </article>
    <article
      class="css-hgavva"
      x="7"
      y="0"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title:  Title
            </h1>
          </header>
        </div>
      </a>
    </article>
    <article
      class="css-2t6kni"
      x="7"
      y="8"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
               Teambesprechung vom 23.05.
            </h1>
          </header>
        </div>
      </a>
    </article>
    <article
      class="css-swar48"
      x="0"
      y="4"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title:  Title
            </h1>
          </header>
        </div>
      </a>
    </article>
    <article
      class="css-1oxqzs1"
      x="3"
      y="6"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title:  Title
            </h1>
          </header>
        </div>
      </a>
    </article>
  </div>
</DocumentFragment>
`;

exports[`Teaser Grid Flex Block should render WithShowLoad 1`] = `
<DocumentFragment>
  <div
    class="css-dkdiq6"
  >
    <article
      class="css-1p0ks0j"
      x="0"
      y="0"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
               Teambesprechung vom 23.05.
            </h1>
          </header>
          <p
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Lead
          </p>
        </div>
      </a>
    </article>
    <article
      class="css-1aj06cw"
      x="3"
      y="0"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title:  Title
            </h1>
          </header>
          <p
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Lead
          </p>
        </div>
      </a>
    </article>
    <article
      class="css-hgavva"
      x="7"
      y="0"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title:  Title
            </h1>
          </header>
          <p
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Lead
          </p>
        </div>
      </a>
    </article>
    <article
      class="css-2t6kni"
      x="7"
      y="8"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
               Teambesprechung vom 23.05.
            </h1>
          </header>
          <p
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Lead
          </p>
        </div>
      </a>
    </article>
    <article
      class="css-swar48"
      x="0"
      y="4"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title:  Title
            </h1>
          </header>
          <p
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Lead
          </p>
        </div>
      </a>
    </article>
    <article
      class="css-1oxqzs1"
      x="3"
      y="6"
    >
      <a
        class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineNone css-9lisvk-MuiTypography-root-MuiLink-root"
        href="https://example.com"
      >
        <div
          class="css-rd719m"
        >
          <img
            alt="An image description"
            class="css-1qnf57e"
            height="500"
            srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
            title="An image title"
            width="500"
          />
          <header
            class="css-0"
          >
            <h1
              class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
            >
              Pre Title:  Title
            </h1>
          </header>
          <p
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            Lead
          </p>
        </div>
      </a>
    </article>
  </div>
</DocumentFragment>
`;
